src/panels/lovelace/sections/hui-grid-section.ts

```diff
@@ -215,6 +215,14 @@ export class GridSection extends LitElement implements LovelaceSectionElement {
           grid-column: span var(--column-size, 4);
         }
 
+        .card:has(> *) {
+          display: block;
+        }
+
+        .card:not(:has(> *:not([hidden]))) {
```

Comment:
Why is it reversed? Why not
```suggestion
        .card:has(> *[hidden]) {
```